chore: update dependencies and maintenance release 0.7.2

yesoreyeram opened the pull request:
- [x] Updated frontend deps
- [x] Updated backend plugin sdk `grafana-plugin-sdk-go`
- [x] Update create plugin config
- [x] Pinned github workflow actions
- [x] Fixed typecheck issues
- [x] Bumped version to 0.7.2 for release

yesoreyeram committed:
updated create-plugin config

yesoreyeram committed:
typecheck fix

yesoreyeram committed:
pinned frontend deps

yesoreyeram committed:
pinned workflow versions

yesoreyeram committed:
updated changelog and version

yesoreyeram committed:
fixed tag in workflow

zoltanbedi commented on .github/workflows/update-create-plugin.yml:
```diff
@@ -13,4 +13,5 @@ jobs:
     runs-on: ubuntu-latest
     steps:
-      - uses: grafana/plugin-actions/create-plugin-update@main
+      - uses: grafana/plugin-actions/create-plugin-update@create-plugin-update/v2.0.1
```
This workflow now requires a token.

yesoreyeram replied:
https://github.com/grafana/grafana-csv-datasource/pull/375/changes/28a57f74a11232f7d5caccc24556291cb863f572

zoltanbedi reviewed (commented):
Just one nit but otherwise lgtm

yesoreyeram committed:
updated create plugin update workflow

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/grafana/grafana-csv-datasource?pullRequest=375) <br/>All committers have signed the CLA.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana-csv-datasource?pullRequest=375) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana-csv-datasource?pullRequest=375) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana-csv-datasource?pullRequest=375) it.</sub>

yesoreyeram committed:
replaced custom time zone picker with grafana ui component (#376)

Replaced custom timezone picker with Grafana UI component ( and so
removed the moment-timezone dependency )
